util/string/expression_parser.cpp

```diff
@@ -88,7 +88,7 @@ bool get_number_token(const std::string_view text, unsigned int &pos, Token &out
 	}
 
 	// Decimal part
-	double f;
+	double f = 0;
```

Comment:
If this default value is necessary, that means the logic following it found an incorrect format... though that's not really handled at the moment. I wonder if compilers generating this warning are still able to optimize this away, because the fact they emit that warning in the first place sounds like they can't...